thunder/transforms/materialization.py

```diff
@@ -13,6 +13,83 @@
     from thunder.core.module import ThunderModule
 
 
+def module_init_from_original_module_init(
+    transform: MaterializationTransform, tm: ThunderModule, *, non_persistent_only: bool = False
+):
+    if non_persistent_only:
+        non_persistent_buffers_set = set()
+        for name, submodule in tm._model.named_modules():
+            prefix = f"{name}." if name else name
+            non_persistent_buffers_set.update(f"{prefix}{bname}" for bname in submodule._non_persistent_buffers_set)
+
+        def neg_filter(n):
+            return n not in non_persistent_buffers_set
```

Comment:
could we have a more explanatory name?

Reply:
I renamed `neg_filter` to `should_skip_materialization`.